runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -83,10 +72,30 @@ protected ILogData removeFromQueue(NavigableSet<Long> queue) {
         if (thisRead != null) {
             // In the case that the immediately requested address (thisRead) is not available,
             // read it along with a batch of subsequent addresses.
-            List<Long> batchRead = getBatch(thisRead, queue);
-            ILogData ld = readRange(thisRead, batchRead);
 
-            if (queue == getCurrentContext().readQueue) {
+            ILogData ld;
+
+            try {
+                // The read range has a special behaviour, it attempts to read the address of interest,
+                // and in the case it is not in the cache, it reads a batch of addresses
+                // that will be further required, returning only the data for the address of interest.
+
+                // Because addresses are requested in increasing order and system relies on prefix trim,
+                // we force trimmedExceptions to be thrown by lower layers, and handle ignoreTrimmed at this layer.
+                // Note that lower layers will cache the valid entries, to optimize read performance.
+                ld = readRange(thisRead, queue, false);
```

Comment:
If readRange is always used with ignoreTrimmed to be false, why not remove this parameter?
Also readRange doesn't sound like a good name, as it returns a single value.

Reply:
Yes, you're right I've changed the name and signature to be more suggestive.